src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -270,7 +318,96 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
+const ImageSelector = ({ control, field, item, selectImage }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [imageInfoText, setImageInfoText] = useState('');
+
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
+
+  return (
+    <>
+      <Input
+        {...item}
+        control={control}
+        errorMessage={errorMessage}
+        hidden
+        validate
+        name={name}
+        value={value}
+      />
+      <RegularText smallest placeholder>
+        {infoText}
+      </RegularText>
+
+      {value ? (
+        <>
+          <WrapperRow center spaceBetween>
+            <Image source={{ uri: value }} style={styles.image} />
+
+            <TouchableOpacity
+              onPress={() => {
+                onChange('');
+                setErrorMessage('');
+                setImageInfoText('');
+              }}
+            >
+              <Icon.Trash color={colors.error} size={normalize(16)} />
+            </TouchableOpacity>
+          </WrapperRow>
+          {!!imageInfoText && <RegularText smallest>{imageInfoText}</RegularText>}
+        </>
+      ) : (
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { uri, type } = await selectImage();
+            const { size } = await FileSystem.getInfoAsync(uri);
+            const errorMessage = imageErrorMessageGenerator({
+              size,
+              imageType: uri.substr(uri.length - 4)
```

Comment:
on a different place it is `- 3` and here `- 4`? as said there, we need a more safe regular expression to determine the image type. and both places must use the same.

Reply:
If I use -3, I will get the `peg` string for `jpeg` images.